src/index.js

```diff
@@ -98,6 +104,22 @@ class PeerId {
       throw new Error('not valid Id')
     }
   }
+
+  /*
+   * Check if this PeerId instance is valid (privKey -> pubKey -> Id)
+   */
+  isValid (callback) {
+    // TODO Needs better checking
```

Comment:
The currently validation isn't really checking if you can generate the same pubkey from the privkey set and consequently the b58 hash. Doing this requires libp2p-crypto operations which need to be async so that we use the WebCrypto API.